### ISSUE TITLE
Service History Pattern Accessibility Review

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/vets-design-system-documentation#5089]
### Product information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [x] Team label, product label and feature label (if applicable) have been added to this issue.

### URL for testing
https://staging.va.gov/mock-simple-forms-patterns/service-branch

### Who completed the use of color and color contrast test?

@jeana-adhoc 

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
   - Controls in combo box have [contrast ratio of 5.57:1](https://webaim.org/resources/contrastchecker/?fcolor=686868&bcolor=FFFFFF) 
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [N/A] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")
- [N/A] (Mobile app only) Color contrast ratios are met for all supported dark, contrast, and inverted color modes
- [N/A] (Mobile app only) All elements retain their meaning for all supported dark, contrast, and inverted color modes

### How did color testing go?

Focus indicator continues to fail color contrast, but is tracked here
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3750

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page or screen has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

No issues

### How did axe testing go?

No issues

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] (Web only) All page elements are readable and usable at 200% zoom
- [X] (Web only) All page elements are readable and usable at 300% zoom
- [X] (Web only) All page elements are readable and usable at 400% zoom
- [N/A] (Mobile app only) All page elements are readable and usable when device text is at the maximum possible size

### How did content zoom and reflow testing go?

No issues

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [X] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

There is a keyboard trap issue happening in the component.  When I type something that is also an "opt group" item, like "Army" several unexpected things happen.
1. The optgroup is duplicated
2. One of the duplicated optgroups is now keyboard focusable, I arrow down, it receives focus ring, but after that I cannot use the arrow keys to select any other item as I would expect. Keyboard is now trapped. The select still scrolls, but focus stays on one of the duplicated opt groups.

https://github.com/user-attachments/assets/fbbfbfec-a43b-457b-a375-436fe76106ee

This is also happening in the main component in storybook. To test:
1. Go to https://design.va.gov/storybook/iframe.html?globals=&args=&id=uswds-va-combo-box--option-groups&viewMode=story
2. Type "e" (this at least gets the Optgroup Vegetable selected and other items) 
3. Vegetable as an opt group displays twice.
4. When you arrow down to one of them, keyboard gets trapped.

### Do you have any other results to share?

1. I also tested this with Mac Voice Control and it works (combobox is clunky but that's voice control issues not the component)

2. In the examples, I would make sure the hint text, label, and opt group selection aligns. The second one says "army" in the label, mentions coast guard and space force in the hint text (and drop down options)

<img width="521" height="347" alt="Image" src="https://github.com/user-attachments/assets/5463d3fa-868c-40ea-b12d-ddfb82811393" />

3. To be consistent, I would remove the placeholder text in the second option. We recommend not using placeholder text throughout VADS.

**Comment by jeana-adhoc** [comment]
# Summary findings

See details above.

Should:
- Remove placeholder text
- Ensure hint text matches the label

Must: 
- Fix opt group bug that causes keyboard trap
  - https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5113
